lib/src/model/accessor.dart

```diff
@@ -17,57 +16,61 @@ import 'package:dartdoc/src/warnings.dart';
 
 /// Getters and setters.
 class Accessor extends ModelElement implements EnclosedElement {
-  GetterSetterCombo enclosingCombo;
+  /// The combo ([Field] or [TopLevelVariable] containing this accessor.
+  /// Initialized by the combo's constructor.
+  late final GetterSetterCombo enclosingCombo;
 
-  Accessor(PropertyAccessorElement element, Library library,
+  Accessor(PropertyAccessorElement? element, Library? library,
       PackageGraph packageGraph,
-      [ExecutableMember /*?*/ originalMember])
+      [ExecutableMember? originalMember])
       : super(element, library, packageGraph, originalMember);
 
   @override
-  CharacterLocation get characterLocation {
-    if (element.nameOffset < 0) {
-      assert(element.isSynthetic, 'Invalid offset for non-synthetic element');
+  CharacterLocation? get characterLocation {
+    if (element!.nameOffset < 0) {
+      assert(element!.isSynthetic, 'Invalid offset for non-synthetic element');
       // TODO(jcollins-g): switch to [element.nonSynthetic] after analyzer 1.8
       return enclosingCombo.characterLocation;
     }
     return super.characterLocation;
   }
 
   @override
-  PropertyAccessorElement get element => super.element;
+  PropertyAccessorElement? get element =>
+      super.element as PropertyAccessorElement?;
 
   @override
-  ExecutableMember get originalMember => super.originalMember;
+  ExecutableMember? get originalMember =>
+      super.originalMember as ExecutableMember?;
 
-  Callable _modelType;
-  Callable get modelType => _modelType ??=
-      modelBuilder.typeFrom((originalMember ?? element).type, library);
+  Callable? _modelType;
+  Callable get modelType => (_modelType ??= modelBuilder.typeFrom(
+      (originalMember ?? element)!.type, library!) as Callable?)!;
 
-  bool get isSynthetic => element.isSynthetic;
+  bool get isSynthetic => element!.isSynthetic;
 
   SourceCodeRenderer get _sourceCodeRenderer =>
       packageGraph.rendererFactory.sourceCodeRenderer;
 
-  GetterSetterCombo _definingCombo;
+  GetterSetterCombo? _definingCombo;
   // The [enclosingCombo] where this element was defined.
-  GetterSetterCombo get definingCombo {
+  GetterSetterCombo? get definingCombo {
```

Comment:
This could benefit from the `late final` pattern, as every access of definingCombo seems to immediately null-check it.